pkg/odo/cli/component/create.go

```diff
@@ -269,7 +269,9 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 		co.interactive = true
 	}
 
-	co.LocalConfigInfo, err = config.NewLocalConfigInfo(co.componentContext)
+	// this populates the LocalConfigInfo as well
+	co.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
```

Comment:
Point to think: we have two ways to init context, one which creates the app i.e. `NewContextCreatingAppIfNeeded`  other that doesn't `NewContext`. Now I am creating the app if needed.
Now some would argue that - post new workflow, all cluster interactions should happen on `odo push` only and I agree to a certain degree. But here an API call moves to the `create` portion which would remove that 250ms from push and its barely noticeable for create. So felt like we can keep this optimisation?

Reply:
IMO, it's an optimization if we're reducing from 250ms to something less. But by moving this from `odo push` to `odo create`, we're simply doing this in a different place.

IMO, if it's agreed (TBH, I'm not aware of the new workflow you've mentioned) to keep cluster interactions to `odo push`, then moving this `odo create` is disobeying it for no apparent reason (is 250ms so noticeable in `odo push`?).  I'm guessing there's a context behind taking a decision to put certain things at certain place. 

This is just a personal opinion. I'm sure others would have their own. :wink:

Reply:
resolved